### PR TITLE
Updated documentation for Parser

### PR DESCRIPTION
**Pull request by deathbeam**
Added documentation for `Parser.parseExpect`, `Parser.noMatch` and `Parser.unexpected`

### CODE REVIEW

File: src/hxparse/Parser.hx [hunk 1]
```diff
@@ -128,15 +132,24 @@ class Parser<S:TokenSource<Token>, Token> {
...
+	/**
+		Throws `Unexpected` exception, which contains last matched position and token
+	**/
 	inline function unexpected():Dynamic {
 		throw new Unexpected(peek(0), stream.curPos());
 	}
-
+	
+	/**
+		Macro thar processes and returns the result of `switch`.
```
Comment (Simn) [inline src/hxparse/Parser.hx]: Can you fix the `thar` typo please?

File: src/hxparse/Parser.hx [hunk 2]
```diff
@@ -128,15 +132,24 @@ class Parser<S:TokenSource<Token>, Token> {
 			unexpected();
 		}
 	}
-
+	
+	/**
+		Throws `NoMatch` exception, which contains last matched position and token
```
Comment (Simn) [inline src/hxparse/Parser.hx]: Missing dot at the end.

File: src/hxparse/Parser.hx [hunk 3]
```diff
@@ -128,15 +132,24 @@ class Parser<S:TokenSource<Token>, Token> {
...
+	**/
 	inline function noMatch() {
 		return new NoMatch(stream.curPos(), peek(0));
 	}
-
+	
+	/**
+		Throws `Unexpected` exception, which contains last matched position and token
```
Comment (Simn) [inline src/hxparse/Parser.hx]: Same here

**Comment by Simn** [comment]
Thanks a lot! Just found some small nitpicks, could you fix this?

**Comment by deathbeam** [comment]
Done :)

**Comment by Simn** [comment]
Thanks!

**Pull request merged**